In [1]:
import numpy as np
import pandas as pd

In [2]:
#VISA
current_share_price_visa = 323.57
eps_visa = 10.74
book_value_per_share_visa = 17.40
EBITDA_visa = 25.214
market_cap_visa = 552.47

In [3]:
#ICICI
current_share_price_icici = 26.17
eps_icici = 1.68
book_value_per_share_icici = 6.01
EBITDA_icici = 0.00
market_cap_icici = 92.04

In [4]:
#SNOWFLAKE
current_share_price_snowflake = 238.26
eps_snowflake = 3.52
book_value_per_share_snowflake = 18.32
EBITDA_snowflake = -1.088
market_cap_snowflake = 84.34

In [10]:
PE_visa = current_share_price_visa / eps_visa
PE_icici = current_share_price_icici/ eps_icici
PE_snowflake = current_share_price_snowflake/ eps_snowflake
print(f"The P/E ratio for Visa is {PE_visa} dollars")
print(f"The P/E ratio for ICICI Bank is {PE_icici} dollars")
print(f"The P/E ratio for Snowflake is {PE_snowflake} dollars")

The P/E ratio for Visa is 30.12756052141527 dollars
The P/E ratio for ICICI Bank is 15.577380952380954 dollars
The P/E ratio for Snowflake is 67.6875 dollars


In [12]:
PB_visa = current_share_price_visa / book_value_per_share_visa
PB_icici = current_share_price_icici/ book_value_per_share_icici
PB_snowflake = current_share_price_snowflake/ book_value_per_share_snowflake
print(f"The P/B ratio for Visa is {PB_visa} dollars")
print(f"The P/B ratio for ICICI Bank is {PB_icici} dollars")
print(f"The P/B ratio for Snowflake is {PB_snowflake} dollars")

The P/B ratio for Visa is 18.595977011494256 dollars
The P/B ratio for ICICI Bank is 4.354409317803661 dollars
The P/B ratio for Snowflake is 13.005458515283841 dollars


In [15]:
ev_by_ebidta_visa = market_cap_visa / EBITDA_visa
ev_by_ebidta_snowflake = market_cap_snowflake / EBITDA_snowflake
print(f"The EV/EBITDA ratio for Visa is {ev_by_ebidta_visa} dollars")
print(f"The EV/EBITDA ratio for Snowflake is {ev_by_ebidta_snowflake} dollars")
print("EBITDA is not a useful metric for evaluating a bank, hence EBITDA of ICICI Bank is 0")

The EV/EBITDA ratio for Visa is 21.91123978741969 dollars
The EV/EBITDA ratio for Snowflake is -77.51838235294117 dollars
EBITDA is not a useful metric for evaluating a bank, hence EBITDA of ICICI Bank is 0


In [19]:
df = pd.DataFrame({
    "Company": ["Visa", "ICICI", "Snowflake"],
    "PE": [PE_visa, PE_icici, PE_snowflake],
    "PB": [PB_visa, PB_icici, PB_snowflake],
    "EV/EBITDA": [ev_by_ebidta_visa, 0, ev_by_ebidta_snowflake]
}
    )
df = df.round(2)
df

,Company,PE,PB,EV/EBITDA
0,Visa,30.13,18.60,21.91
1,ICICI,15.58,4.35,0.00
2,Snowflake,67.69,13.01,-77.52


The P/E ratio for ICICI Bank is the cheapest of all 3 which is 15.57 dollars
The P/B ratio for ICICI Bank is the cheapest of all 3 which is 4.34 dollars
EV/EBITDA ratio is positive for Visa Inc. (21.91) dollars, 0 for ICICI Bank and -77.51 dollars for Snowflake. This means that VISA Inc. makes a profit today and the investors invest heavily as they pay a premium on this expecting future growth. Snowflake has a negative EV/EBIDTA (-77.51 dollars) which means that the company is actively burning cash.